In [1]:
#import the libraries
import pandas as pd
import scipy 
import numpy as np
from scipy import stats
import statsmodels.api as sm
from statsmodels.formula.api import ols

In [2]:
df = pd.read_csv("C:/Users/kiran/Desktop/Assignments/Hypothesis Testing/LabTAT.csv")

In [3]:
df.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [4]:
L1 = df.iloc[:,0] 
L2 = df.iloc[:,1] 
L3 = df.iloc[:,2] 
L4 = df.iloc[:,3]
L1.dtypes

dtype('float64')

In [5]:
#all_scores = L1 + L2 + L3 + L4
all_scores = np.concatenate([L1, L2, L3, L4])

In [6]:
Labs = (['L1'] * len(L1) + ['L2'] * len(L2) + ['L3'] * len(L3) + ['L4'] * len(L4))

In [9]:
data = pd.DataFrame({'Labs': Labs, 'Score': all_scores})
data

Labs   Score
0     L1  185.35
1     L1  170.49
2     L1  192.77
3     L1  177.33
4     L1  193.41
..   ...     ...
475   L4  172.68
476   L4  177.64
477   L4  170.27
478   L4  150.87
479   L4  162.21

[480 rows x 2 columns]

In [8]:
data.groupby('Labs').mean()

score
Labs            
L1    178.361583
L2    178.902917
L3    199.913250
L4    163.682750

In [10]:
result = ols('Score ~ Labs', data = data).fit()
table = sm.stats.anova_lm(result)
print(table)

             df         sum_sq       mean_sq           F        PR(>F)
Labs        3.0   79979.167089  26659.722363  118.704217  2.115671e-57
Residual  476.0  106904.608903    224.589515         NaN           NaN


#As F-Ratio(118.70) is greater than the F-Critical value, we reject the Null Hypothesis